File: app/src/main/java/com/condorlabs/popularmovies/di/ActivityBuilderModule.java
package com.condorlabs.popularmovies.di;

import com.condorlabs.popularmovies.home.MainActivity;
import com.condorlabs.popularmovies.home.MoviesHomeModule;
import com.condorlabs.popularmovies.moviedetail.MovieDetailActivity;
import com.condorlabs.popularmovies.moviedetail.MovieDetailModule;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityBuilderModule {

    @ContributesAndroidInjector(modules = MoviesHomeModule.class)
    abstract MainActivity mainActivity();

    @ContributesAndroidInjector(modules = MovieDetailModule.class)
    abstract MovieDetailActivity movieDetailActivity();


}

File: app/src/main/java/com/condorlabs/popularmovies/di/ActivityBuilderModule.kt
package com.condorlabs.popularmovies.di

import com.condorlabs.popularmovies.home.MainActivity
import com.condorlabs.popularmovies.home.MoviesHomeModule
import com.condorlabs.popularmovies.moviedetail.MovieDetailActivity
import com.condorlabs.popularmovies.moviedetail.MovieDetailModule

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityBuilderModule {

    @ContributesAndroidInjector(modules = [MoviesHomeModule::class])
    internal abstract fun mainActivity(): MainActivity

    @ContributesAndroidInjector(modules = [MovieDetailModule::class])
    internal abstract fun movieDetailActivity(): MovieDetailActivity


}
